src/aws-cpp-sdk-core/source/utils/crypto/CRC64.cpp

```diff
@@ -0,0 +1,33 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/core/utils/crypto/CRC64.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/core/utils/crypto/Factories.h>
+
+using namespace Aws::Utils::Crypto;
+
+CRC64::CRC64(): m_hashImpl(CreateCRC64Implementation())
+{
+}
+
+HashResult CRC64::Calculate(const Aws::String& str)
+{
+    return m_hashImpl->Calculate(str);
+}
+
+HashResult CRC64::Calculate(Aws::IStream& stream)
+{
+    return m_hashImpl->Calculate(stream);
+}
+
+void CRC64::Update(unsigned char* buffer, size_t bufferSize)
+{
```

Comment:
null check? 
also constness of the ptr,  unsigned char* const buffer ?

Reply:
cant because the [original API is not](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/include/aws/core/utils/crypto/Hash.h#L22-L51), and we cannot change it without breaking backwards compatibilty.